src/VisualStudio/Core/Def/Implementation/InfoBar/VisualStudioInfoBarService.cs

```diff
@@ -46,13 +43,12 @@ public void ShowInfoBar(string message, params InfoBarUI[] items)
             ThisCanBeCalledOnAnyThread();
 
             // We can be called from any thread since errors can occur anywhere, however we can only construct and InfoBar from the UI thread.
-            _foregroundNotificationService.RegisterNotification(() =>
+            System.Threading.Tasks.Task.Run(async () =>
```

Comment:
```suggestion
            ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
```

Reply:
📝 Note that unlike Task.Run, JoinableTaskFactory.RunAsync invokes its callback synchronously. This means you can move the `BeginAsyncOperation` inside the callback with a using declaration statement, and not need the `CompletesAsyncOperation` call.

Reply:
noice